uriget/uriget.go

```diff
@@ -244,25 +244,45 @@ func (o *options) getGit(ctx context.Context, u *url.URL) ([]byte, error) {
 func (o *options) getOci(ctx context.Context, u *url.URL) ([]byte, error) {
 	ref, err := registry.ParseReference(u.Host + u.Path)
 	if err != nil {
-		return nil, fmt.Errorf("can't parse artifact URL into a valid reference: %w", err)
+		return nil, fmt.Errorf("invalid artifact URL: %w", err)
 	}
-	store, err := oci.New(o.tempDir)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create OCI layout store: %w", err)
+	if ref.Reference == "" {
+		ref.Reference = "latest"
 	}
+	specifiedFile := strings.TrimPrefix(u.Fragment, "#")
 	remoteRepo, err := remote.NewRepository(ref.String())
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to remote repository: %w", err)
+		return nil, fmt.Errorf("connection to remote repository failed: %w", err)
+	}
+	_, rc, err := remoteRepo.Manifests().FetchReference(ctx, ref.Reference)
+	if err != nil {
+		return nil, fmt.Errorf("manifest fetch failed: %w", err)
 	}
-	tag := "latest"
-	if ref.Reference != "" {
-		tag = ref.Reference
+	defer rc.Close()
+	var manifest v1.Manifest
+	if err = json.NewDecoder(rc).Decode(&manifest); err != nil {
+		return nil, fmt.Errorf("manifest decode failed: %w", err)
+	}
+	var selectedLayer *v1.Descriptor
+	for _, layer := range manifest.Layers {
+		title := layer.Annotations[v1.AnnotationTitle]
+		if (specifiedFile != "" && title == specifiedFile) ||
+			(specifiedFile == "" && strings.HasSuffix(title, ".yaml")) {
+			selectedLayer = &layer
+			break
+		}
 	}
-	manifestDescriptor, err := oras.Copy(ctx, remoteRepo, tag, store, tag, oras.DefaultCopyOptions)
+	if selectedLayer == nil {
+		return nil, fmt.Errorf("no matching .yaml file found in layers")
+	}
+	_, rc, err = remoteRepo.Blobs().FetchReference(ctx, selectedLayer.Digest.String())
 	if err != nil {
-		return nil, fmt.Errorf("failed to pull OCI image: %w", err)
+		return nil, fmt.Errorf("blob fetch failed: %w", err)
 	}
-
-	o.logger.Printf("Pulled OCI image: %s with manifest descriptor: %v", u.String(), manifestDescriptor.Digest)
-	return []byte(manifestDescriptor.Digest), nil
+	defer rc.Close()
+	raw, err := io.ReadAll(rc)
```

Comment:
Please use the `readLimited` here like we do in the other fetchers to keep limit behavior consistent :)

Reply:
for sure! :)